### ISSUE TITLE
target-size rule should be applicable to nodes in the body element

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3739]
When you put a link directly in the body of a page, target-size considers it inapplicable:

```
<html>
  <a href="#">link</a>
</html>
```

Weirdly, this doesn't seem to happen with `button`. It looks like this has something to do with the `isInTextBlock` method.

**Comment by padmavemulapati** [comment]
Validated with the latest build,
for the test snippet:
```
<a role="link" aria-label="play" tabindex="0">
    <button tabindex="-1" style="margin:1px;">Play</button>
  </a>
```
![image](https://user-images.githubusercontent.com/47409579/197195016-40ff4b81-4ff5-4f7f-9a09-8944d4789914.png)


Also one more observation:
If we wrap this `href` within `div` it will be resulting `incomplete`
```
<div> <a role="link" aria-label="play" tabindex="0">
    <button tabindex="-1" style="margin:1px;">Play</button>
  </a> </div>
```
![image](https://user-images.githubusercontent.com/47409579/197195301-5ee93c59-bc3f-40b8-82ee-c5f11215d13f.png)

**Comment by arpitbharadwaj1** [comment]
Checked this against current axe-core (4.10.2) with the exact repro:

```html
<a href="#">link</a>
<button>button</button>
```

Both elements are now flagged as `target-size` **violations** — the asymmetry is gone. (Also just found and fixed a related already-resolved-but-untested `target-size` false positive in #4295 → #5343, so this rule area seems to have had real fixes land since these were filed, likely as part of the ongoing target-size epic in #5225.)

Since both elements now get the same, correct treatment, this looks resolved. Given the fix already ships, I don't think a dedicated regression test is needed here the way #4295 warranted one — the current behavior is just ordinary rule output, not a special edge case — but happy to add one if you'd still like coverage locking in the anchor-directly-in-body case specifically.

**Comment by straker** [comment]
This took a bit of time to debug. When using just the provided sample code and then loading axe-core on the page from devtools, running target-size produces a passes result. However when loading the code and adding a script tag as well to run axe, the results turn in to inapplicable.

```html
<html>
  <a href="#">link</a>
  <script src="/axe.js"></script>
  <script>
  (async () => {
    const res = await axe.run({ runOnly: 'target-size' })
    console.log(res);
    // inapplicable
  })();
  </script>
</html>
```

Debugging the code it turns out we're adding the text content of the script tag in the [`walkDOM` function of `is-in-text-block`](https://github.com/dequelabs/axe-core/blob/develop/lib/commons/dom/is-in-text-block.js#L39C3-L39C14). So the presence of the inline script tag changes the results.
